### PR TITLE
[Youtube] Added channel view count and subscriber count badges

### LINKED ISSUE #6111: [Youtube] subscriber counter/ total view time
:clipboard: **Description**

Service: Youtube

:link: **Data**

-Youtube API   (I don't know exactly how)

:microphone: **Motivation**

Maks the information on an internetsite visible to show how "good" the youtube channel is.

### PR DESCRIPTION
**Pull request by tarun7singh**
Added Channel view count and subscriber count badge.

- Moved Youtube base to youtube video base.
- Added Youtube channel subscriber badge
- Added Youtube channel total view count badge
- Added tests for both badges

Fixes #6111

### CODE REVIEW

**Comment by tarun7singh** [comment]
I had added another issue's code in this PR by mistake. Reverted it with a revert commit.

File: services/youtube/youtube-channel-base.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        viewCount: nonNegativeInteger,
+        subscriberCount: nonNegativeInteger,
+        hiddenSubscriberCount: Joi.boolean().required(),
+        videoCount: nonNegativeInteger,
+      }).required(),
+    })
+  ),
+}).required()
+
+class YouTubeChannelBase extends BaseJsonService {
```
Comment (PyvesB) [inline services/youtube/youtube-channel-base.js:29]: As far as I can tell, this file is almost identical to `YouTubeVideoBase`. Could this please be refactored back to a single `YouTube` base? Some of the few differences between the two classes can be phased out:
* using the hostname, `www.googleapis.com`, as we've hitting the same underlying API in both cases.
* using the same check for the not found case. `if (!json.items || json.items.length === 0)` should cover both the video and channel responses.

The only remaining differences will be a few string values and the API schema - if you pass `{ videoId, channelId }` to the `handle` method, you should be able to infer the right values depending on which value is actually defined.

File: services/youtube/youtube-channel-base.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  pageInfo: Joi.object({
+    totalResults: nonNegativeInteger,
+    resultsPerPage: nonNegativeInteger,
+  }).required(),
+  items: Joi.array().items(
+    Joi.object({
+      statistics: Joi.object({
+        viewCount: nonNegativeInteger,
+        subscriberCount: nonNegativeInteger,
+        hiddenSubscriberCount: Joi.boolean().required(),
```
Comment (PyvesB) [inline services/youtube/youtube-channel-base.js:22]: No need to validate on `hiddenSubscriberCount` and `videoCount` if we're not actually using them.

File: services/youtube/youtube-subscribers.service.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  static get examples() {
+    const preview = this.render({
+      statistics: { subscriberCount: 14577 },
+      channelId: 'UCTNq28Ah5eyDgsYOA0oHzew',
+    })
+    // link[] is not allowed in examples
+    delete preview.link
+    return [
+      {
+        title: 'YouTube Subscriber Counts',
```
Comment (PyvesB) [inline services/youtube/youtube-subscribers.service.js:20]: Could this please be changed to `YouTube Channel Subscribers`? As you'll notice in #6304, the YouTube compliance team was extremely picky about some of the wording used in our UI, I don't want to give them any opportunity to claim that it's not clear that the badge relates to channels.

File: services/youtube/youtube-subscribers.service.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+module.exports = class YouTubeSubscribes extends YouTubeChannelBase {
+  static route = {
+    base: 'youtube/channel/subscribers',
+    pattern: ':channelId',
+  }
+
+  static get examples() {
+    const preview = this.render({
+      statistics: { subscriberCount: 14577 },
+      channelId: 'UCTNq28Ah5eyDgsYOA0oHzew',
```
Comment (PyvesB) [inline services/youtube/youtube-subscribers.service.js:14]: Could a programming/IT/coding/developer channel be used instead in all examples, including the ones in `YouTubeChannelViews`? The Apache Foundation's channel (`UCLDJ_V9KUOdOFSbDvPfGBxw`) could be a good candidate.

Our [contributing guidelines](https://github.com/badges/shields/blob/master/CONTRIBUTING.md) state that "_Shields.io hosts integrations for services which are primarily used by developers or which are widely used by developers_". A lot of our social badges are somewhat borderline, I would at least like to send the right message in the examples of said badges.

File: services/youtube/youtube-channel-views.tester.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+'use strict'
+
+const t = (module.exports = require('../tester').createServiceTester())
+const { noToken } = require('../test-helpers')
+const { isMetric } = require('../test-validators')
+const noYouTubeToken = noToken(require('./youtube-channel-views.service'))
+
+t.create('channel view count')
```
Comment (PyvesB) [inline services/youtube/youtube-channel-views.tester.js:8]: All the newly added tests are clear and comprehensive, thanks for that. 👍🏻

**Review by PyvesB — CHANGES_REQUESTED** [review]
Thanks for the contribution @tarun7singh ! 👍🏻 

The timing of this pull request is a bit unfortunate. I've just completed quite a tedious compliance process with the YouTube team (see #6276 for a bit more details). One of their conclusive notes is the following:

> Please note that the quota extension has been granted ONLY for the use-case(s) mentioned in your quota extension request application and we may change your quota based on your actual quota usage as well as on your continued compliance with the YouTube API Services Terms of Service at any time.

Obviously I hadn't mentioned the new channel badges as they didn't yet exist at the time, but equally one could argue that it's an extension of the same use case. I suggest we move ahead with this, but one my review comments will be particularly important to address to make sure we don't break compliance.

File: services/youtube/youtube-subscribers.service.js (hunk 4 above)
Comment (tarun7singh) [reply services/youtube/youtube-subscribers.service.js:14]: fixesd

**Comment by tarun7singh** [comment]
@PyvesB Completed all the requested changes.

**Comment by PyvesB** [comment]
@tarun7singh thanks for promptly addressing my review comments. Things mostly look good, though I think `YouTubeBase` can further be improved. Here's what I came up with after a bit of experimentation:
https://github.com/badges/shields/commit/55ecd9416dce02da4b5f83fe2824c48c124d65bc

Let me know what you think and feel free to cherry-pick that commit back onto your PR branch. Otherwise, happy to approve as it is, I don't want to be too picky, so it's your call. 😉

**Comment by tarun7singh** [comment]
@PyvesB Yes your code looks much mor econcise but I was not able to successfully cherry pick this commit. can you please do it or tell me how to do it?

**Comment by PyvesB** [comment]
Try the following:
```
git remote add fork-pyvesb https://github.com/PyvesB/shields
git fetch fork-pyvesb
git cherry-pick 55ecd9416dce02da4b5f83fe2824c48c124d65bc
```

Alternatively, give me write permissions to your fork so that I can push the commit.

**Comment by tarun7singh** [comment]
@PyvesB Added your commti to the PR.

**Review by PyvesB — APPROVED** [review]
Thanks. Looks all good to me, let's get this merged! :+1:

**Pull request merged**